webapp/plugins/vue-directives.js

```diff
@@ -7,7 +7,10 @@ export default ({ app }) => {
       // Focus the element
       Vue.nextTick(() => {
         if (binding.value !== false) {
-          el.focus()
+          const target = el instanceof HTMLInputElement ? el : el.querySelector('input')
+          if (target) {
+            target.focus()
+          }
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-26T18:22:40Z" title="Thursday, March 26th 2020, 7:22:40 pm +01:00">Mar 26, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -7,7 +7,14 @@ export default ({ app }) => {
       // Focus the element
       Vue.nextTick(() => {
         if (binding.value !== false) {
-          el.focus()
+          let target = el
+          const isInput = target instanceof HTMLInputElement
+          if (!isInput) {
+            target = el.querySelector('input')
+          }
+          if (target && typeof target.focus === 'function') {
+            target.focus()
+          }
```


is the fix to the issue you were facing in the tests? do you have a link to where you found this fix?
Also, would need to have a closer look at this.

Reply:
> [<img alt="rbeer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/rbeer) **Authored by [rbeer](https://github.com/rbeer)**
_<time datetime="2020-03-30T07:03:35Z" title="Monday, March 30th 2020, 9:03:35 am +02:00">Mar 30, 2020</time>_

---


`v-focus` has no tests.
`v-focus` just existed; it was not used by any component, so I took the liberty to make it target <input\> elements.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-31T12:04:43Z" title="Tuesday, March 31st 2020, 2:04:43 pm +02:00">Mar 31, 2020</time>_

---


I meant the issue with `Vue.nextTick()` not doing what we expected in the test you wrote.

Reply:
> [<img alt="rbeer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/rbeer) **Authored by [rbeer](https://github.com/rbeer)**
_<time datetime="2020-04-01T04:55:23Z" title="Wednesday, April 1st 2020, 6:55:23 am +02:00">Apr 1, 2020</time>_

---


Then no, this vue-directives `v-focus` change is unrelated to that.

Like you suggested, I had a look at other use cases of Vue.nextTick() in the tests and noticed that none of those other instances waits for the actual result of a function. It's always just used like
```js
wrapper.find('button').emit('click');
await Vue.nextTick();
expect(getFooAsync).toHaveBeenCalled()
```
i.e. not testing to what the Promise resolved, merely whether its producing function had been called.

So I went to the [jest documentation](https://jestjs.io/docs/en/timer-mocks) and found that jest manages its own micro-queue for timers.

There is a package called flush-promises which I think is mentioned in the vue docs, but I didn't want to install another package. Turns out, that it is [already in use](https://github.com/Human-Connection/Human-Connection/blob/f13eb29ad342392edee61b82c02a126a5fdbae3b/webapp/pages/settings/my-social-media.spec.js#L64).

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-04-01T05:57:45Z" title="Wednesday, April 1st 2020, 7:57:45 am +02:00">Apr 1, 2020</time>_

---


yes, that's great! nice catch!